src/Uno.Toolkit.UI/Controls/ZoomContentControl/ZoomContentControl.cs

```diff
@@ -0,0 +1,482 @@
+	{
+		if (!(IsAllowedToWork && _capturedPointerId > 0 && IsPanAllowed)) return;
+
+		var currentPosition = e.GetCurrentPoint(this).Position;
+		_movementDirection = (currentPosition.X > _lastPosition.X, currentPosition.Y > _lastPosition.Y);
+		_lastPosition = currentPosition;
+
+		if (CanMoveIn(_movementDirection))
+		{
+			e.Handled = true;
+			var pointerPoint = e.GetCurrentPoint(this);
+			var position = pointerPoint.Position;
+			var deltaX = position.X - _referencePosition.X;
+			var deltaY = position.Y - _referencePosition.Y;
+			TryUpdateOffsets(deltaX, deltaY);
+			_referencePosition = position;
+		}
+	}
+
+	private void OnPointerWheelChanged(object sender, PointerRoutedEventArgs e)
+	{
+		if (!IsAllowedToWork) return;
+
+		var pointerPoint = e.GetCurrentPoint(this);
+		var pointerProperties = pointerPoint.Properties;
+
+		var changeRatio = GetZoomDelta(pointerProperties);
+
+		if (
+#if IS_WINUI
+			pointerPoint.PointerDeviceType == PointerDeviceType.Mouse &&
+#else
+			pointerPoint.PointerDevice.PointerDeviceType == PointerDeviceType.Mouse &&
+#endif
+			e.KeyModifiers.HasFlag(Windows.System.VirtualKeyModifiers.Control) &&
+			IsZoomAllowed)
+		{
+			e.Handled = true;
+
+			var relativeX = (pointerPoint.Position.X - HorizontalOffset) / ZoomLevel;
+			var relativeY = (pointerPoint.Position.Y - VerticalOffset) / ZoomLevel;
+
+			ZoomLevel *= changeRatio;
+
+			HorizontalOffset = pointerPoint.Position.X - (relativeX * ZoomLevel);
+			VerticalOffset = pointerPoint.Position.Y - (relativeY * ZoomLevel);
+			return;
+		}
+
+		if (e.KeyModifiers.HasFlag(Windows.System.VirtualKeyModifiers.Shift))
+		{
+			var deltaX = GetPanDelta(pointerProperties);
+			TryUpdateOffsets(deltaX, 0);
+			return;
+		}
+
+		var deltaY = GetPanDelta(pointerProperties);
+		TryUpdateOffsets(0, deltaY);
+	}
+
+	private double GetZoomDelta(PointerPointProperties pointerProperties)
+	{
+		var delta = pointerProperties.MouseWheelDelta * ScaleWheelRatio;
+		return 1 + delta;
+	}
+
+	private double GetPanDelta(PointerPointProperties pointerProperties)
+	{
+		var delta = pointerProperties.MouseWheelDelta * PanWheelRatio;
+		return delta;
+	}
+
+	private void TryUpdateOffsets(double deltaX, double deltaY)
+	{
+		if ((deltaX > 0 && CanScrollLeft()) ||
+			(deltaX < 0 && CanScrollRight()))
+		{
+			var offset = HorizontalOffset + deltaX;
+			var max = HorizontalMaxScroll * ZoomLevel;
+			HorizontalOffset = Math.Clamp(offset, 0, max);
+		}
+
+		if ((deltaY > 0 && CanScrollUp()) ||
+			(deltaY < 0 && CanScrollDown()))
+		{
+			var offset = VerticalOffset + deltaY;
+			var max = VerticalMaxScroll * ZoomLevel;
+			VerticalOffset = Math.Clamp(offset, 0, max);
+		}
+	}
+
+	// Public methods
+	public void Initialize()
+	{
+		if (_presenter?.Content is FrameworkElement { } fe)
+		{
+			fe.UpdateLayout();
+		}
+
+		ResetZoom();
+		ResetOffset();
+		CenterContent();
+	}
+
+	public void ResetZoom() => ZoomLevel = 1;
+
+	public void ResetOffset()
```

Comment:
Does this need to be public?

Reply:
`ResetZoom` needs to be public, I have set the `ResetOffset` and `RemoveOffset` to private